### ISSUE TITLE
App Redirects to Main Screen When Downloading with High Storage Usage

### ISSUE THREAD

**Issue by wshsirin8** [avluis/Hentoid#1200]
**Hentoid Version**:  Hentoid v1.20.5 (755)

**Device/Android Version**:  Redmi K70 Pro / Android 14

**Issue details / Repro steps**:  I encountered an issue where, after tapping the triangular download button on the download list page, the app redirected me back to the main screen. At the time, my device’s storage usage was 400GB out of 1TB.

After reviewing issue #1198, I deleted some duplicate comics, which temporarily resolved the issue, allowing me to download comics normally. I would like to know if there's a way to avoid this problem in the future.

**Stack trace / LogCat**:
```ruby
latest-crash log : begin
Hentoid ver: 1.20.5 (755)
API: 34
Device: 23117RK66C
Start : 2024-11-14T16:58:11.737696Z
End : 2024-11-14T16:58:11.737795Z (00:00)
-----
Could not put
io.objectbox.exception.DbFullException: Could not put
	at io.objectbox.Cursor.collect004000(Native Method)
	at me.devsaki.hentoid.database.domains.ContentCursor.put(SourceFile:159)
	at me.devsaki.hentoid.database.domains.ContentCursor.put(SourceFile:24)
	at io.objectbox.Box.put(SourceFile:411)
	at me.devsaki.hentoid.database.ObjectBoxDB.updateContentStatus(SourceFile:175)
	at me.devsaki.hentoid.database.ObjectBoxDAO.updateContentStatus(SourceFile:382)
	at me.devsaki.hentoid.viewmodels.QueueViewModel.unpauseQueue(SourceFile:192)
	at me.devsaki.hentoid.fragments.queue.QueueFragment.onCreateView$lambda$2(SourceFile:202)
	at me.devsaki.hentoid.fragments.queue.QueueFragment.$r8$lambda$s4Frb7ErmAoC9v4Qv0GEcYc23sM(SourceFile:0)
	at me.devsaki.hentoid.fragments.queue.QueueFragment$$ExternalSyntheticLambda12.onClick(SourceFile:0)
	at android.view.View.performClick(View.java:7798)
	at android.view.View.performClickInternal(View.java:7775)
	at android.view.View.-$$Nest$mperformClickInternal(Unknown Source:0)
	at android.view.View$PerformClick.run(View.java:31136)
	at android.os.Handler.handleCallback(Handler.java:958)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:224)
	at android.os.Looper.loop(Looper.java:318)
	at android.app.ActivityThread.main(ActivityThread.java:8719)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:561)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1013)

latest-crash log : end
```

@AVnetWS/admin-team

**Comment by RobbWatershed** [comment]
Thank you for posting the crash log. 

I'm afraid your confusing three different problems :
- **RAM usage** (#1198) : too many apps opened at once and/or little RAM available + high resource consumption

- **Storage usage** : too many files on your device; usually comes with a warning notification from both Android _and_ Hentoid. If you say you're using 400GB over 1TB (40%), you're definitely not in that situation.

- **Database usage** : too many books and/or too many images inside your Hentoid library. Hentoid has a limit on how many books / images it can store (around 40K-50K books), and your error log tells you've hit that limit. You can check that in Settings > Storage > Storage Management > View usage statistics ("Database" on the bottom of the screen).

Max size is currently set to 2GB. I guess I can raise it to 3GB, but I don't guarantee smooth performance with such an enormous amount of data. Would you be OK with that option?

**Comment by wshsirin8** [comment]
First of all, I sincerely apologize for my immature and confusing initial description. Here is the complete sequence of events I experienced:

1. I initially noticed that downloads failed despite having sufficient storage space.
2. After reviewing issue #1198, I confirmed that my device's RAM was more than adequate.
3. Finally, I checked the log file and discovered that the issue was caused by the database limit. However, I am not sure why this limitation was triggered.

Secondly, I would like to clarify my situation:

1. I specifically purchased this high-capacity phone to use apps like Hentoid, downloaded from GitHub.  
2. This device is dedicated solely to reading comics and does not run any other apps in the background.

Lastly, I am uncertain about the performance impacts of increasing the database size limit. Therefore, I cannot provide a definitive answer at this moment regarding the proposed change.

Thank you for your patience and understanding. I appreciate your assistance in resolving this issue.

**Comment by wshsirin8** [comment]
![IMG_20241116_001954](https://github.com/user-attachments/assets/5671973a-b3f4-471f-944d-f94f0c1dc7cf)

**Comment by wshsirin8** [comment]
![Screenshot_2024-11-16-00-20-26-212_com miui home](https://github.com/user-attachments/assets/31bcfcbf-e12f-41a8-8513-3d16dfd46aef)

**Comment by RobbWatershed** [comment]
No problem man. Solving an issue means understanding it in the first place.

I think the solution is indeed to raise the database size limit. I'll do that for next release 👍 

That being said, filling the database with only 12K books looks abnormal to me. I've seen the app handle much more than that.

By any chance, are you using the merge feature to create big books with many pages? Or maybe are you collecting books with 1000s of pages such as game CGs?

**Comment by wshsirin8** [comment]
Regarding your questions, here’s some additional information:

1. I’ve only used the **merge feature** once, to combine multiple chapters of a comic tagged as “webtoon.” These comics are in a vertical scrolling format and usually have hundreds of pages, with some reaching 1000 or even 2000 pages.

2. For other types of content, like “image sets,” I’ve downloaded very few.

3. My library currently contains around 300 webtoons.

**Comment by RobbWatershed** [comment]
Makes sense. Plus Hitomi has super long image URLs now (e.g. `https://aa.hitomi.la/webp/1732086001/1156/2d47d5adbe803b13bda6c5664b61b61d25bb2809ca6206681024c04472f9d844.webp`)

I'll raised the database size limit to 3GB for next release. After you update, please check on the storage screen if that limit has indeed been raised. If it hasn't, you should reinstall the app from scratch (NB : your books won't be lost).

**Comment by RobbWatershed** [comment]
Fix is available on today's v1.20.6
